### PR TITLE
Impl hideable tasks lists

### LINKED ISSUE #26: Make lists collapsible or hideable
We should provide a way to hide some lists, by collapsing or just not displaying them and creating a menu to choose which list to show.

### LINKED ISSUE #27: Change members avatars colors

### LINKED ISSUE #40: Setup release-please
Setup release-please for this repo.

https://github.com/googleapis/release-please

### PR DESCRIPTION
**Pull request by swouf**
Allow the lists to be hidden.

Close #26
Close #27
Close #40 

- [x] Also fix problem in cypress workflow

### CODE REVIEW

File: src/components/main/Task.tsx [hunk 1]
```diff
@@ -95,78 +101,86 @@ const Task = (props: TaskProps): JSX.Element => {
...
-      data: {
-        type,
-      },
-      disabled: dialogOpen,
-    });
-
-  const transformation: React.CSSProperties = {
-    transform: CSS.Translate.toString(transform),
-  };
+  const renderTask = (isDragging: boolean): JSX.Element => (
```
Comment (spaenleh) [inline src/components/main/Task.tsx:104]: It looks like you could refactor this into its own component and rename the current component `TaskWrapper` or something. I read recently that is is not a good idea to have nested render functions ([see in Tao of React](https://alexkondov.com/tao-of-react/#avoid-nested-render-functions)).

File: src/components/main/Task.tsx [hunk 2]
```diff
@@ -95,78 +101,86 @@ const Task = (props: TaskProps): JSX.Element => {
...
+      }}
+    >
+      <CardHeader
+        avatar={
+          <AvatarGroup max={3}>
+            {members.map((member) => (
+              <Avatar
+                key={member}
+                sx={{
+                  bgcolor: `${getMemberColor(member)}`,
```
Comment (spaenleh) [inline src/components/main/Task.tsx:125]: You seem to use the `members.find()` multiple times for the same member. 
Solutions: 
- find your member from memberList once at the start of your `map`
- change `memberList` to be an object with the `memberId` as keys so you can access the member in constant time with `memberObject[member.id]`. When you will need to iterate over your memberObject you can always use the `Object.entries(memberObject)` method to get an array that you can map over.

File: src/components/main/TasksListMinimized.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <Button
+    ref={ref}
+    variant="outlined"
+    endIcon={<ArrowForwardIosIcon />}
+    onClick={onShow}
+    sx={{ width: '100%', mt: 2 }}
+  >
+    <Badge badgeContent={tasks.size} color="secondary">
+      <Typography variant="h2">{title}</Typography>
+    </Badge>
```
Comment (spaenleh) [inline src/components/main/TasksListMinimized.tsx:31]: You could factor this out into `TaskTitle` and re-use it here and in your `TaskList`.

File: src/components/views/TasksManager.tsx [hunk 4]
```diff
@@ -47,6 +54,16 @@ const TasksManager = (): JSX.Element => {
 
   const { appSettingArray } = useAppSettingContext();
 
+  const [hiddenLists, setHiddenLists] = useState<Map<string, boolean>>(
+    Map([
+      [TASK_LABELS.TODO, false],
+      [TASK_LABELS.IN_PROGRESS, false],
+      [TASK_LABELS.COMPLETED, false],
+    ]),
+  );
```
Comment (spaenleh) [inline src/components/views/TasksManager.tsx:63]: Could this work ?
```suggestion
  const [hiddenLists, setHiddenLists] = useState<{[key: string]: boolean}>(
    {
      [TASK_LABELS.TODO]: false,
      [TASK_LABELS.IN_PROGRESS]: false,
      [TASK_LABELS.COMPLETED]: false,
    },
  );
```

File: src/components/views/TasksManager.tsx [hunk 5]
```diff
@@ -137,44 +154,122 @@ const TasksManager = (): JSX.Element => {
...
 
+  const handleDragStart = (event: DragStartEvent): void => {
+    setActiveTask(tasks.find((ta) => ta.id === event.active.id) ?? null);
+  };
   const renderTasksList = (
     title: string,
     label: string,
     add = false,
   ): JSX.Element => {
     // eslint-disable-next-line react/destructuring-assignment
```
Comment (spaenleh) [inline src/components/views/TasksManager.tsx:168]: Remove ?

File: src/components/views/TasksManager.tsx [hunk 6]
```diff
@@ -137,44 +154,122 @@ const TasksManager = (): JSX.Element => {
...
+            {activeTask ? (
+              <Task
+                key={0}
+                task={activeTask}
+                updateTask={(): void => {
+                  throw new Error('Function not implemented.');
+                }}
+                deleteTask={(): void => {
+                  throw new Error('Function not implemented.');
+                }}
```
Comment (spaenleh) [inline src/components/views/TasksManager.tsx:267]: Finish implementation ? Or are these not used ?

**Review by spaenleh — CHANGES_REQUESTED** [review]
🗒️ 🔮 Looks good ! 
I have left some suggestions for refactoring, and small questions, otherwise looks nice ! 
I am not sure if there are are any visual changes in this PR, if yes, would be nice to document these with screen shots or small gifs/vidéos. For example I don't think i know how it looks with the minimized task lists on the side ?

File: src/components/views/TasksManager.tsx (hunk 6 above)
Comment (swouf) [reply src/components/views/TasksManager.tsx:267]: These are not used. But I could just put empty functions.

Comment (spaenleh) [reply src/components/views/TasksManager.tsx:267]: Yes, or better, make the props optional, if the component can work without them.

Comment (swouf) [reply src/components/views/TasksManager.tsx:267]: Is it better in your opinion?

File: src/components/views/TasksManager.tsx [hunk 7]
```diff
@@ -160,18 +158,22 @@ const TasksManager: FC = () => {
   const handleDragStart = (event: DragStartEvent): void => {
     setActiveTask(tasks.find((ta) => ta.id === event.active.id) ?? null);
   };
+
+  const handleHide = (label: string, hide: boolean): void => {
+    const newHiddenLists = {
+      ...hiddenLists,
+      [label]: hide,
+    };
+    setHiddenLists(newHiddenLists);
```
Comment (spaenleh) [inline src/components/views/TasksManager.tsx:167]: Although your implementation is not wrong, we might prefer a different approach:

You can provide an updater function to the SetState method, you get the previous state for free and this solves a lot of issues with out-of-date state 😉 
```suggestion
    setHiddenLists((prevState) => ({
      ...prevState,
      [label]: hide,
    }));
```

**Review by spaenleh — APPROVED** [review]
🧸 Nice refactor ! I suggest you a tiny change, but the rest looks good, I like how you took the opportunity to improve 🦾

File: src/components/views/TasksManager.tsx (hunk 6 above)
Comment (swouf) [reply src/components/views/TasksManager.tsx]: I did change the component in the end. Thanks for the help.

**Comment by swouf** [comment]
Sorry for bothering you, but could I ask for one more review please 😄

**Review by spaenleh — APPROVED** [review]
This looks good, I am just a bit surprised that you can use `t` inside the de-structuring call... It does not show a warning ? If not, then all is good to me ! Don't worry about asking for too many reviews ahah 💪

**Pull request merged**